src/ui/blockUi-editor.js

```diff
@@ -16,9 +16,63 @@ https://raw.githubusercontent.com/fluid-project/sjrk-story-telling/master/LICENS
     // the base blockUi for editing individual blocks, contains shared elements
     fluid.defaults("sjrk.storyTelling.blockUi.editor", {
         gradeNames: ["sjrk.storyTelling.blockUi"],
+        model: {
+            // reorder buttons are disabled depending on the block's order:
+            // if the block is first in the order, the "up" button is disabled
+            // if the block is last in the order, the "down" button is disabled
+            moveBlockDownDisabled: "{block}.model.lastInOrder",
+            moveBlockUpDisabled: "{block}.model.firstInOrder"
+        },
+        modelListeners: {
+            "moveBlockDownDisabled": {
+                func: "{that}.setButtonDisabled",
+                args: ["{that}.dom.moveBlockDownButton", "{change}.value"],
+                excludeSource: ["init"],
```

Comment:
Should be able to avoid partnering this with the "onReady" duplicate listeners, especially given you are on the FLUID-6145-era Infusion - what is the result if you simply remove the excludeSource annotation here?

Reply:
I believe the difference is invisible to the user. The model values will be set early on when the DOM isn't ready (the buttons aren't rendered yet), so nothing happens when `.prop()` is called. The listeners hanging off of `onReadyToBind` are to set the buttons once the markup is rendered, and the modelListeners will handle any subsequent changes. I'm ambivalent about excluding init, at this point, though I suppose we might have some barely-perceptible performance gains by not handling it on init.

Can you please elaborate on what FLUID-6145 is allowing me to do or not do?

Reply:
One of the main improvements in the FLUID-6148/FLUID-6145 revision ensures that the timing of the "init" model transaction is fully predictable with respect to a component's "onCreate" - that is, it always comes before it. This should mean that the function of your duplicate listeners in "onReadyToBind" on lines 65-74 should be handlable by the model listener here as part of "init" - especially if you can move whatever it is that the component needs to be "ready" into the component's resources - the only such thing should be a template loading, as far as I can see, since everything following that should be synchronous

Reply:
We discussed this issue a bit [in IRC](http://irc-logs.fluidproject.org/%23fluid-work/%23fluid-work.2020-06-18.log) today, and came to the decision to defer this work for another Jira. I've filed [SJRK-364](https://issues.fluidproject.org/browse/SJRK-364) to capture your suggestions, please feel free to amend it if it isn't accurate :)

I'll be adding a comment in this part of the code linking to both this review item and the new Jira.